torch/_C/__init__.pyi.in

```diff
@@ -743,8 +743,10 @@ def __set_forward_AD_enabled(enabled: _bool) -> None: ...
 def __is_forward_AD_enabled() -> _bool: ...
 def _register_default_hooks(pack_hook: Callable, unpack_hook: Callable) -> None: ...
 def _reset_default_hooks() -> None: ...
-def _enter_python_mode(cls: Type) -> None: ...
-def _exit_python_mode() -> None: ...
+def _set_torch_function_mode(cls: Type) -> None: ...
```

Comment:
Should cls be union[TorchFunctionMode, Type] ? Also I am not sure if you applied the same treatment for torch_dispatch mode (still going through this)

Reply:
Oh good point...it errored in the second PR probably because it wasn't that union. I wonder if we'll end up with a circular dependency if this needs the Mode types? Well, let's see!